streams/shared/src/main/scala/zio/stream/experimental/ZStream.scala

```diff
@@ -128,4 +128,37 @@ object ZStream extends Serializable {
         }
       } yield Control(pull, Command.noop)
     }
+
+  /**
+   * Creates a stream from an iterable collection of values
+   *
+   * @tparam A the iterable value type
+   * @return a stream which emits values from iterable
+   */
+  def fromIterable[A](as: => Iterable[A]): ZStream[Any, Nothing, Any, Unit, A] =
+    ZStream {
+      Managed.effectTotal {
+        Control(
+          IO.succeedNow(as.iterator).flatMap(iter => if (iter.hasNext) IO.succeed(iter.next()) else Pull.endUnit),
```

Comment:
It looks like you could just write `fromIterator(as.iterator)` for this function.

Reply:
Yes, that'd be preferable. I think the current version does `fromChunk(Chunk.fromIterable(as))` which should be faster in most common cases.

Reply:
That makes sense, I'll implement it using chunk.